### PR TITLE
Add processModuleFilter and remove postProcessModules

### PR DESCRIPTION
**Pull request by cliberal**
**Summary**
remove postProcessModules hook since it is not working now. Add a processModuleFilter hook to skip some modules before code generating. see #210 

**Test plan**
pass

### CODE REVIEW

File: docs/Configuration.md [hunk 1]
```diff
@@ -104,4 +104,4 @@ These options are only useful with React Native projects.
...
 | `postProcessBundleSourcemap`    | `PostProcessBundleSourcemap`                                           | An optional function that can modify the code and source map of the bundle before it is written. Applied once for the entire bundle.        |
 | `getModulesRunBeforeMainModule` | `(entryFilePath: string) => Array<string>`                             | An array of modules to be required before the entry point. It should contain the absolute path of each module.                              |
-| `postProcessModules`            | `(modules: Array<Module>, entryFiles: Array<string>) => Array<Module>` | A function that can change the resulting modules output.                                                                                    |
+| `processModuleFilter`            | `(module: Array<Module>) => boolean` | A function that can change the resulting modules output.                                                                                    |
```
Comment (rafeca) [inline docs/Configuration.md]: Can you change the description to: `A filter function to discard specific modules from the output`?

File: packages/metro/src/DeltaBundler/Serializers/plainJSBundle.js [hunk 2]
```diff
@@ -42,6 +43,7 @@ function plainJSBundle(
     ...getAppendScripts(entryPoint, graph, options),
   ]
     .filter(isJsModule)
+    .filter(options.processModuleFilter)
```
Comment (rafeca) [inline packages/metro/src/DeltaBundler/Serializers/plainJSBundle.js]: This needs to be done to [each of the serializers](https://github.com/facebook/metro/tree/master/packages/metro/src/DeltaBundler/Serializers) to have consistent results no matter the type of output that the user is generating (it's specially important to update the sourcemap serializers to not end up with incorrect sourcemaps).

**Comment by cliberal** [comment]
Thanks for review
1. I did not change `getAllFiles.js` since it does't seem to be used.
2. There some redundant code but I'm not sure whether can I remove it.

```js
// sourceMapObject.js

function fullSourceMapObject(
  pre: $ReadOnlyArray<Module<>>,
  graph: Graph<>,
  options: {|
    +excludeSource: boolean,
    +processModuleFilter: (module: Module<>) => boolean,
  |},
): BabelSourceMap {
  const modules = [...pre, ...graph.dependencies.values()]
    .filter(isJsModule)
    .filter(options.processModuleFilter) // this line can be removed if we don't use it in `_sourceMapForURL` function below
    .map(module => {
      return {
        ...getJsOutput(module).data,
        path: module.path,
        source: options.excludeSource ? '' : module.getSource(),
      };
    });

  return fromRawMappings(modules).toMap(undefined, {
    excludeSource: options.excludeSource,
  });
}

// Server.js
async _sourceMapForURL(reqUrl: string): Promise<MetroSourceMap> {
    const options: DeltaOptions = this._getOptionsFromUrl(reqUrl);

    const {graph, prepend} = await this._getGraphInfo(options, {
      rebuild: false,
    });

    return sourceMapObject(prepend, graph, {
      excludeSource: options.excludeSource,
      processModuleFilter: this._config.serializer.processModuleFilter,
    });
 }
```

**Comment by rafeca** [comment]
> 1. I did not change getAllFiles.js since it does't seem to be used.

`getAllFiles()` sadly is being used by [`Server.getOrderedDependencyPaths()`](https://github.com/facebook/metro/blob/master/packages/metro/src/Server.js#L238), which sadly is being used by the `react-native` cli (there's [a cli command](https://github.com/facebook/react-native/blob/master/local-cli/dependencies/dependencies.js#L52) to get all the files in the bundle).

> 2. There some redundant code but I'm not sure whether can I remove it.

The `sourceMapObject()` method is being used by the Server [symbolication entrypoint](https://github.com/facebook/metro/blob/master/packages/metro/src/Server.js#L857). This is an http entrypoint `http://localhost:8081/symbolicate` that allows simulators to display the original file/line number of errors/warnings (you may have noticed that RN development error boxes initially display the raw line number and after a small delay it shows the original file/line/column: this is because the app is making a request to the symbolication entrypoint).

**Pull request closed without merging**